chia/wallet/nft_wallet/nft_wallet.py

```diff
@@ -247,16 +239,11 @@ async def puzzle_solution_received(self, coin_spend: CoinSpend, peer: WSChiaConn
 
         self.log.info("Adding a new NFT to wallet: %s", child_coin)
         # all is well, lets add NFT to our local db
-        parent_coin = None
-        confirmed_height = None
-        coin_states: Optional[List[CoinState]] = await self.wallet_state_manager.wallet_node.get_coin_state(
-            [coin_name], peer=peer
+        parent_coin = uncurried_nft.coin_state.coin
+        confirmed_height = (
+            None if uncurried_nft.coin_state.spent_height is None else uint32(uncurried_nft.coin_state.spent_height)
         )
 
-        if coin_states is not None:
-            parent_coin = coin_states[0].coin
-            confirmed_height = None if coin_states[0].spent_height is None else uint32(coin_states[0].spent_height)
-
         if parent_coin is None or confirmed_height is None:
```

Comment:
Yes, since the NFT wallet doesn't create any derived puzzle hash, the only code path to the coin_added is the determine_coin_type in the wallet manager. So we expect there must be a parent coin data passed in.